docs/modules/pwm2.md

```diff
@@ -0,0 +1,195 @@
+# PWM Module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2019-02-12 | [fikin](https://github.com/fikin) | [fikin](https://github.com/fikin) | [pwm2.c](../../../app/modules/pwm2.c)|
+
+Module to generate PWM impulses to any of the GPIO pins. 
+
+PWM is being generated by software using soft-interrupt TIMER1 FRC1. This timer is used by other modules like ws2812 and pwm, but only one of them can use it at given time. Also some parts of OS might mask the interrupt for awhile.
```

Comment:
The note that the OS may mask interrupts for a while is a point that I think ought to be made more prominent. When I was working on the original PWM code, there was a fair amount of complexity to try and deal with interrupts being masked off (and then deciding what to do about it). What does the actual pulse train of a 50% 200kHz signal look like in the presence of (say) running an http server which serves up small blobs of data?

Reply:
i haven't tested it that far yet ;)  on relatively idling env i've checked up to 100kHz and haven't spotted side effects.

Reply:
if you don't object, i'll close this until some more real world feedback is gathered.